src/Behat/Testwork/PathOptions/Printer/ConfigurablePathPrinter.php

```diff
@@ -32,15 +35,55 @@ public function setPrintAbsolutePaths(bool $printAbsolutePaths): void
         $this->printAbsolutePaths = $printAbsolutePaths;
     }
 
+    public function setEditorUrl(?string $editorUrl): void
+    {
+        $this->editorUrl = $editorUrl;
+    }
+
     /**
-     * Conditionally transforms paths to relative.
+     * Conditionally transforms paths to relative and adds editor links if configured.
      */
     public function processPathsInText(string $text): string
     {
-        if ($this->printAbsolutePaths === true) {
-            return $text;
+        // If no editor URL is set, use the original behavior
+        if ($this->editorUrl === null) {
+            if ($this->printAbsolutePaths === true) {
+                return $text;
+            }
+
+            return str_replace($this->basePath . DIRECTORY_SEPARATOR, '', $text);
         }
 
-        return str_replace($this->basePath . DIRECTORY_SEPARATOR, '', $text);
+        // Search for paths in the text
+        $basePathPattern = preg_quote($this->basePath . DIRECTORY_SEPARATOR, '/');
```

Comment:
Our text is not a single text with a path, it is a text that can contain multiple paths (for example in an error stack trace) immersed in the text, so we need to loop over any path found and apply our settings